### PR TITLE
Siamese Network(Task 7) s4636418 Yicheng Jia

### PR DESCRIPTION
**Pull request by YichengJia**
Hi Teaching Teams,
I got an extension until today so please don't mark me as late submission.
Thank you!

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:  Good
- Clear flow of information
- Algortihm and usage explained well
- Some plots provided

Commit messages:  Poor, minimal and not detailed

Code: 
- design is OK
- well commented
- doesn't seem to have performed data augmentation or any hyper-parameter tuning

Functionality/Performance: 
- **only performing similarity detection between pairs of images, no classification**
- achieves good accuracy for similarity detection with the siamese 

General comments:
- **algorithm solves the respective problem only half-way**
- you need to have added a classifier that can classify an individual image as AD or NC with their features extracted by the trained siamese network

**Comment by YichengJia** [comment]
Hi Course Team,
I have modified my codes and fixed the structure. Please check my new codes later.
Thank you very much for any help you provided this semester!

**Comment by shakes76** [comment]
# Marking
Marked as per the extended due date and changes after which aren't necessarily allowed to contribute to grade for fairness
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Partially Solves problem no classification in initial PR -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no augmentation -1
Difficulty: Hard
## Commit Log
Meaningful commit messages minimal -1
Progressive commits used
## Documentation
ReadMe acceptable, no refs -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required, done already -2 +2
Request Description good
